Add NULL checks to getBuffer calls and catch a potential file leak
http://bugs.icu-project.org/trac/ticket/12378
--- icu4c-77_1/icu/source/extra/uconv/uconv.cpp
+++ icu4c-77_1/icu/source/extra/uconv/uconv.cpp
@@ -445,7 +445,10 @@ return -1; // wait for actual further co
         }
     }
 
-    const char16_t *u = s.getBuffer(), *limit = u + s.length();
+    const UChar *u = s.getBuffer();
+    if (u == NULL)
+        return -1;
+    const UChar *limit = u + s.length();
     char16_t c;
 
     while (u < limit) {
@@ -1348,6 +1351,7 @@ convert them. */
 
     cf.setBufferSize(bufsz);
 
+    try {
     if(remainArgv < remainArgvLimit) {
         for (iter = remainArgv; iter != remainArgvLimit; iter++) {
             if (!cf.convertFile(
@@ -1367,6 +1371,11 @@ ) {
             goto error_exit;
         }
     }
+    }
+    catch (int e)
+    {
+        goto error_exit;
+    }
 
     goto normal_exit;
 error_exit:
